### ISSUE TITLE
Several problems on CPU mining after fork.

### ISSUE THREAD

**Issue by hugopnabais** [fireice-uk/xmr-stak#2293]
Several problems started after latest for 1788000:

1º Xmr-stak 2.10.0 is crashing in all my computers with old CPUs that CPU mine.
happens on several different servers.
Example of message: 
 Problem Event Name:	APPCRASH
  Application Name:	xmr-stak.exe
  Application Version:	0.0.0.0
  Application Timestamp:	5c818ea6
  Fault Module Name:	xmr-stak.exe
  Fault Module Version:	0.0.0.0
  Fault Module Timestamp:	5c818ea6
  Exception Code:	c000001d
  Exception Offset:	0000000000063fce
  OS Version:	6.3.9600.2.0.0.272.7

2º Mining on modern CPUs after a while slows down a lot and I have to restart xmr-stak to get performance back!

# Basic information
1º crash problem example CPU intel Xeon E5520
1º crash problem example CPU AMD Optron 6172
1º crash problem example CPU intel Q9500

2º slowdown problem example intel i7-4790k
2º slowdown problem example intel i7-8550
2º slowdown problem example intel i7-3520M

Version 2.10.0 was working fine before fork!

# Compile issues
  - Which OS do you use?
Windows 10 Windows 2012 server.

**Comment by psychocrypt** [comment]
Do you compiled the miner or do you use the pr-compiled binaries?

**Comment by rickbb** [comment]
Not trying to hijack the thread, but have the same issue with intel Pentium cpu. I set the cpu to null in the config.txt and stak then runs with gpus, but there is other issues. I will put in a separate thread.

**Comment by hugopnabais** [comment]
pre-compiled binaries

**Comment by tevador** [comment]
> 2º Mining on modern CPUs after a while slows down a lot and I have to restart xmr-stak to get performance back!

Confirmed, having the same issue. CPU hashrate randomly drops from ~510 H/s to ~110 H/s and I have to restart to fix it.

Linux, v2.10.0-56d2770, self-compiled.

**Comment by Spudz76** [comment]
` Exception Code: c000001d` is an illegal instruction, meaning you've hit AES or AVX2 code within a CPU that doesn't have it.

set aes false in config.txt and asm off in cpu.txt

**Comment by ceruleanfirm** [comment]
Hi,

v2.10.0-56d2770
self-compiled

:~/xmr-stak$ uname -a
Linux o0 4.15.0-46-generic #49~16.04.1-Ubuntu SMP Tue Feb 12 17:45:24 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
:~/xmr-stak$ grep CPU /proc/cpuinfo 
model name	: Intel(R) Core(TM) i7-7700K CPU @ 4.20GHz
model name	: Intel(R) Core(TM) i7-7700K CPU @ 4.20GHz
model name	: Intel(R) Core(TM) i7-7700K CPU @ 4.20GHz
model name	: Intel(R) Core(TM) i7-7700K CPU @ 4.20GHz
model name	: Intel(R) Core(TM) i7-7700K CPU @ 4.20GHz
model name	: Intel(R) Core(TM) i7-7700K CPU @ 4.20GHz
model name	: Intel(R) Core(TM) i7-7700K CPU @ 4.20GHz
model name	: Intel(R) Core(TM) i7-7700K CPU @ 4.20GHz
:~/xmr-stak$ grep '^\"' config.txt 
"call_timeout" : 10,
"retry_time" : 30,
"giveup_limit" : 0,
"verbose_level" : 3,
"print_motd" : true,
"h_print_time" : 60,
"aes_override" : true,
"use_slow_memory" : "no_mlck",ing
"tls_secure_algo" : true,
"daemon_mode" : false,
"output_file" : "",
"httpd_port" : 8888,
"http_login" : "",
"http_pass" : "",
"prefer_ipv4" : true,
:~/xmr-stak$ 

I restarted xmr-stak after the fork in order to test changes in nvidia.txt. And then,
21 h/s per cpu thread. 
"can do better ..."
Pretty sure it was normal hashrate before I restarted.
Tried with "aes_override" : null,
same same ...

**Comment by werdwerdus** [comment]
> ` Exception Code: c000001d` is an illegal instruction, meaning you've hit AES or AVX2 code within a CPU that doesn't have it.
> 
> set aes false in config.txt and asm off in cpu.txt

this worked for me on i7-930 and i5-760, thank you

**Comment by ceruleanfirm** [comment]
Thx for reply,
but no.
i7-7700K support AES-NI ...

**Comment by Spudz76** [comment]
Yes that's great, but [Windows can disable AVX](https://superuser.com/a/623738) regardless what your CPU supports which is what is failing (AVX exists but is disabled by OS, which we didn't previously check the software-offswitch).

**Comment by psychocrypt** [comment]
check your bioas settings. aes support can be disabled in the most bios

**Comment by hugopnabais** [comment]
On my i7-4790k (supports AES-NI) I have 4 threads each at around 80KH/s (total 320KH/s) and after a while, it drops to around 20 KH/s per thread for no apparent reason (total 80KH/s).
If I close and open xmr-stak it goes back to 80KH/s per thread.
This happens on several different computers that all support AES-NI.

**Comment by psychocrypt** [comment]
this bug is already solved in the dev branch and will be part of the today
release.

**Comment by hugopnabais** [comment]
Ok, thank you, I was just making sure you guys were aware.

**Comment by ceruleanfirm** [comment]
I've just cloned dev branch and ...
it'ssss aliiiive !
Thx !
